File: 2026-02/2026-02-20/is_valid_trick.py
def is_valid_trick(trick_name):
    groups_dict = re.match(r"^(?P<first_word>\w+) (?P<second_word>\w+)$", trick_name).groupdict()
    valid_first_words = ["Misty", "Ghost", "Thunder", "Solar", "Sky", "Phantom", "Frozen", "Polar"]
    valid_second_words = ["Twister", "Icequake", "Avalanche", "Vortex", "Snowstorm", "Frostbite", "Blizzard", "Shadow"]

    return groups_dict['first_word'] in valid_first_words \
        and groups_dict['second_word'] in valid_second_words

# --- TEST SUITE ---

tests_text = r'''
1. is_valid_trick("Polar Vortex") should return True.
2. is_valid_trick("Solar Icequake") should return True.
3. is_valid_trick("Thunder Blizzard") should return True.
4. is_valid_trick("Phantom Frostbite") should return True.
5. is_valid_trick("Ghost Avalanche") should return True.
6. is_valid_trick("Snowstorm Shadow") should return False.
7. is_valid_trick("Solar Sky") should return False.
'''

import re

tests_regex = re.compile(r"(?P<number>\d+)\.\s(?P<function_call>.+) should return (?P<output>.+?)\.?$", re.MULTILINE)
test_data = [item.groupdict() for item in re.finditer(tests_regex, tests_text)]

def run_tests(test_data):
    print("--------------------------")
    print("🧪 Starting Verification...")
    print("--------------------------")

    fail_count = 0

    for test in test_data:
        function_call_output = eval(test['function_call'])
        test_output = eval(test['output'])
        
        if function_call_output == test_output:
            print(f"{test['number']}.✅ PASS - Function Call:\n{test['function_call']}")
        else:
            print(f"{test['number']}.❌ FAIL - Function Call:\n{test['function_call']}\nExpected:\n{test_output}\nGot:\n{function_call_output}")
            fail_count += 1

    print("----------------------------",
        f"\n⚠️ WARNING: {fail_count}/{len(test_data)} tests FAILED." if fail_count
        else "\n🎉 SUCCESS: All tests PASSED."
    )

run_tests(test_data)

File: 02 February/02-20/isValidTrick.js
function isValidTrick(trickName) {
  const { firstWord, secondWord } = /^(?<firstWord>\w+) (?<secondWord>\w+)$/.exec(trickName).groups;
  const validFirstWords = ["Misty", "Ghost", "Thunder", "Solar", "Sky", "Phantom", "Frozen", "Polar"];
  const validSecondWords = ["Twister", "Icequake", "Avalanche", "Vortex", "Snowstorm", "Frostbite", "Blizzard", "Shadow"];

  return validFirstWords.includes(firstWord) 
    && validSecondWords.includes(secondWord);
}

// --- TEST SUITE ---

const testsText = String.raw`
1. isValidTrick("Polar Vortex") should return true.
2. isValidTrick("Solar Icequake") should return true.
3. isValidTrick("Thunder Blizzard") should return true.
4. isValidTrick("Phantom Frostbite") should return true.
5. isValidTrick("Ghost Avalanche") should return true.
6. isValidTrick("Snowstorm Shadow") should return false.
7. isValidTrick("Solar Sky") should return false.
`;

const testsRegex = /(?<number>\d+)\.\s(?<functionCall>.+) should return (?<output>.+?)\.?$/gm;
const testData = [...testsText.matchAll(testsRegex).map(match => match.groups)];

function runTests(testData) {
  console.log("--------------------------");
  console.log("🧪Starting Verification...");
  console.log("--------------------------");

  let failCount = 0;
  
  testData.forEach(test => {
    const functionCallOutput = eval(test.functionCall);
    const testOutput = eval(test.output);

    const comparison = Array.isArray(testOutput)
      ? arraysEqual(functionCallOutput, testOutput)
      : functionCallOutput === testOutput;

    if (comparison) {
      console.log(`${test.number}.✅PASS - Function Call: ${test.functionCall}`);
    }
    else {
      console.log(`${test.number}.❌FAIL - Function Call: ${test.functionCall}\nExpected: ${testOutput}\nGot: ${functionCallOutput}`);
      ++failCount;
    }
  })

  console.log("----------------------------",
    failCount
    ? `\n⚠️WARNING: ${failCount}/${testData.length} tests FAILED.`
    : "\n🎉SUCCESS: All tests PASSED."
  );
}

function arraysEqual(a, b) {
  if (a.length !== b.length)
    return false;

  for (let i = 0; i < a.length; ++i) {
    if (Array.isArray(a[i])) {
      if (Array.isArray(b[i])) {
        if (!arraysEqual(a[i], b[i]))
          return false;
      }
      else 
        return false;
    }

    else if (a[i] !== b[i])
      return false;
  }

  return true;
}

runTests(testData);